File: Samples/ReadExifMetadata/C#/ReadExifMetadata.cs
using System.Diagnostics;
using System.IO;
using System.Windows.Forms;

using BitMiracle.LibTiff.Classic;

namespace BitMiracle.LibTiff.Samples
{
    public static class ReadExifMetadata
    {
        public static void Main()
        {
            using (Tiff image = Tiff.Open(@"Sample data\dscf0013.tif", "r"))
            {
                if (image == null)
                {
                    MessageBox.Show("Could not open incoming image");
                    return;
                }

                FieldValue[] exifIFDTag = image.GetField(TiffTag.EXIFIFD);
                if (exifIFDTag == null)
                {
                    MessageBox.Show("Exif metadata not found");
                    return;
                }

                int exifIFDOffset = exifIFDTag[0].ToInt();
                if (!image.ReadEXIFDirectory(exifIFDOffset))
                {
                    MessageBox.Show("Could not read EXIF IFD");
                    return;
                }

                using (StreamWriter writer = new StreamWriter("ReadExifMetadata.txt"))
                {
                    for (TiffTag tag = TiffTag.EXIF_EXPOSURETIME; tag <= TiffTag.EXIF_IMAGEUNIQUEID; ++tag)
                    {
                        FieldValue[] value = image.GetField(tag);
                        if (value != null)
                        {
                            for (int i = 0; i < value.Length; i++)
                            {
                                writer.WriteLine("{0}", tag.ToString());
                                writer.WriteLine("{0} : {1}", value[i].Value.GetType().ToString(), value[i].ToString());
                            }

                            writer.WriteLine();
                        }
                    }
                }
            }

            Process.Start("ReadExifMetadata.txt");
        }
    }
}

File: Samples/ReadExifMetadata/VB.NET/ReadExifMetadata.vb
Imports System.Diagnostics
Imports System.IO
Imports System.Windows.Forms

Imports BitMiracle.LibTiff.Classic

Namespace BitMiracle.LibTiff.Samples
    Public NotInheritable Class ReadExifMetadata
        Private Sub New()
        End Sub
        Public Shared Sub Main()
            Using image As Tiff = Tiff.Open("Sample data\dscf0013.tif", "r")
                If image Is Nothing Then
                    MessageBox.Show("Could not open incoming image")
                    Return
                End If

                Dim exifIFDTag As FieldValue() = image.GetField(TiffTag.EXIFIFD)
                If exifIFDTag Is Nothing Then
                    MessageBox.Show("Exif metadata not found")
                    Return
                End If

                Dim exifIFDOffset As Integer = exifIFDTag(0).ToInt()
                If Not image.ReadEXIFDirectory(exifIFDOffset) Then
                    MessageBox.Show("Could not read EXIF IFD")
                    Return
                End If

                Using writer As New StreamWriter("ReadExifMetadata.txt")
                    For tag As TiffTag = TiffTag.EXIF_EXPOSURETIME To TiffTag.EXIF_IMAGEUNIQUEID
                        Dim value As FieldValue() = image.GetField(tag)
                        If value IsNot Nothing Then
                            For i As Integer = 0 To value.Length - 1
                                writer.WriteLine("{0}", tag.ToString())
                                writer.WriteLine("{0} : {1}", value(i).Value.[GetType]().ToString(), value(i).ToString())
                            Next

                            writer.WriteLine()
                        End If
                    Next
                End Using
            End Using

            Process.Start("ReadExifMetadata.txt")
        End Sub
    End Class
End Namespace